test/timeout.js

```diff
@@ -27,69 +45,211 @@ test('timeout option (ETIMEDOUT)', async t => {
 			retry: 0
 		}),
 		{
-			code: 'ETIMEDOUT'
+			...errorMatcher,
+			message: `Timeout awaiting 'request' for 0ms`
 		}
 	);
 });
 
-test('timeout option (ESOCKETTIMEDOUT)', async t => {
+test('timeout option as object (ETIMEDOUT)', async t => {
 	await t.throws(
 		got(s.url, {
-			timeout: reqDelay,
+			timeout: {socket: reqDelay * 2.5, request: 1},
 			retry: 0
 		}),
 		{
-			code: 'ESOCKETTIMEDOUT'
+			...errorMatcher,
+			message: `Timeout awaiting 'request' for 1ms`
 		}
 	);
 });
 
-test('timeout option as object (ETIMEDOUT)', async t => {
+test('socket timeout', async t => {
 	await t.throws(
 		got(s.url, {
-			timeout: {socket: reqDelay * 2.5, request: 0},
+			timeout: {socket: reqTimeout},
 			retry: 0
 		}),
 		{
-			code: 'ETIMEDOUT'
+			instanceOf: got.TimeoutError,
+			code: 'ETIMEDOUT',
+			message: `Timeout awaiting 'socket' for ${reqTimeout}ms`
 		}
 	);
 });
 
-test('timeout option as object (ESOCKETTIMEDOUT)', async t => {
+test('send timeout', async t => {
 	await t.throws(
 		got(s.url, {
-			timeout: {socket: reqDelay * 1.5, request: reqDelay},
+			timeout: {send: 1},
 			retry: 0
 		}),
 		{
-			code: 'ESOCKETTIMEDOUT'
+			...errorMatcher,
+			message: `Timeout awaiting 'send' for 1ms`
 		}
 	);
 });
 
-test('socket timeout', async t => {
+test('response timeout', async t => {
 	await t.throws(
 		got(s.url, {
-			timeout: {socket: reqDelay / 20},
+			timeout: {response: 1},
+			retry: 0
+		}),
+		{
+			...errorMatcher,
+			message: `Timeout awaiting 'response' for 1ms`
+		}
+	);
+});
+
+test('response timeout (slow upload)', async t => {
+	const body = fs.createReadStream('/dev/urandom', {
```

Comment:
Ugh, I forgot about AppVeyor. I'll look for an alternative. Suggestions welcome.

Reply:
I made https://github.com/sindresorhus/random-bytes-readable-stream, but I see you already solved the problem :)

Reply:
That looks handy :-)